stdpopsim/ext/genomic_elements.py

```diff
@@ -0,0 +1,22 @@
+import attr
+
+@attr.s(kw_only=True)
+class GenomicElementType(object):
+    """
+    TODO: docstring.
+
+    See SLiM documentation for initializeGenomicElementType().
+    """
+    mutation_types = attr.ib(default=[stdpopsim.ext.MutationType()], type=list)
+    start_positions = attr.ib(default=None, type=np.array)
+    end_positions = attr.ib(default=None, type=np.array)
+    slim_mutations_fraction = attr.ib(frozen=True)
+
+    @slim_mutations_fraction.default
+    def _sum_proportions(self):
+        return sum([mut_type.proportion for mut_type in self.mutation_types])
+
+    def __attrs_post_init__(self):
```

Comment:
check that all `end_pos > start_pos`

Reply:
`>=` since a segment of 1bp will have start==end

Reply:
oh wait never mind, sorry! yes, you're right, a segment of 1bp will be `[a, a+1)`.